### PR TITLE
Implement git fetch

### PR DESCRIPTION
**Pull request by chaodaiG**
https://github.com/kubernetes/test-infra/pull/23264 uses this function

### CODE REVIEW

File: prow/git/git.go [hunk 1]
```diff
@@ -507,11 +507,21 @@ func (r *Repo) MergeCommitsExistBetween(target, head string) (bool, error) {
 }
 
 // ShowRef returns the commit for a commitlike. Unlike rev-parse it does not require a checkout.
-func (i *Repo) ShowRef(commitlike string) (string, error) {
-	i.logger.Infof("Getting the commit sha for commitlike %s", commitlike)
-	out, err := i.gitCommand("show-ref", "-s", commitlike).CombinedOutput()
+func (r *Repo) ShowRef(commitlike string) (string, error) {
+	r.logger.Infof("Getting the commit sha for commitlike %s", commitlike)
```
Comment (fejta) [inline prow/git/git.go:511]: consider using a field for commitlike

**Review by fejta — APPROVED** [review]
/hold

**Review by fejta — APPROVED** [review]

**Review by cjwagner — APPROVED** [review]
Thanks, I'm surprised this is the first time we're using `fetch`.
/hold

File: prow/git/git.go (hunk 1 above)
Comment (chaodaiG) [reply prow/git/git.go:511]: this PR only renamed `i` to `r` for consistency reason, surprisingly github displays diff like this.

I can fix all the other logger in a separate PR

**Pull request merged**